src/lucky/file_response.cr

```diff
@@ -0,0 +1,81 @@
+class Lucky::FileResponse < Lucky::Response
+  DEFAULT_STATUS = 200
+
+  getter context, path, disposition, filename, debug_message
+
+  def initialize(@context : HTTP::Server::Context,
+                 @path : String,
+                 @content_type : String? = nil,
+                 @disposition : String = "attachment",
+                 @filename : String? = nil,
+                 @status : Int32? = nil,
+                 @debug_message : String? = nil)
+  end
+
+  def print
+    raise Lucky::Exceptions::MissingFile.new("Cannot read file #{path}") unless file_exists?
+
+    set_response_headers
+    context.response.status_code = status
+    File.open(full_path) { |file| IO.copy(file, context.response) }
+  end
+
+  def status
+    @status || context.response.status_code || DEFAULT_STATUS
+  end
+
+  def set_response_headers
+    context.response.content_length = File.size(full_path)
+    context.response.content_type = content_type
+    context.response.headers["Accept-Ranges"] = "bytes"
+    context.response.headers["X-Content-Type-Options"] = "nosniff"
+    context.response.headers["Content-Transfer-Encoding"] = "binary"
+
+    if filename
+      context.response.headers["Content-Disposition"] =
+        %(#{disposition}; filename="#{filename}")
+    else
+      context.response.headers["Content-Disposition"] = disposition
+    end
+  end
+
+  def content_type
+    @content_type || content_type_from_file
+  end
+
+  private def content_type_from_file : String
+    extension = File.extname(path)
+    inferred_content_type = {
+      ".css"   => "text/css",
+      ".gif"   => "image/gif",
+      ".htm"   => "text/html",
+      ".html"  => "text/html",
+      ".ico"   => "image/x-icon",
+      ".jpg"   => "image/jpeg",
+      ".jpeg"  => "image/jpeg",
+      ".js"    => "application/javascript",
+      ".json"  => "application/json",
+      ".mp4"   => "video/mp4",
+      ".otf"   => "application/font-sfnt",
+      ".ttf"   => "application/font-sfnt",
+      ".png"   => "image/png",
+      ".svg"   => "image/svg+xml",
+      ".txt"   => "text/plain",
+      ".webm"  => "video/webm",
+      ".woff"  => "application/font-woff",
+      ".woff2" => "font/woff2",
+      ".xml"   => "application/xml",
+      ""       => "application/octet-stream",
+    }[extension]
+
+    return inferred_content_type
```

Comment:
Instead of the `return` you could remove the local variable and return and it should work the same

Reply:
duh, of course. Remnants of some previous hacking around with this.